src/control/lib/hardware/fabric.go

```diff
@@ -136,7 +314,7 @@ func (s fabricInterfaceMap) update(name string, fi *FabricInterface) {
 		// always possible to add to providers or net interfaces
 		if fi.Providers != nil {
 			if cur.Providers == nil {
-				cur.Providers = common.NewStringSet()
+				cur.Providers = newTestFabricProviderSet()
 			}
 			cur.Providers.Add(fi.Providers.ToSlice()...)
```

Comment:
Nice, so the FabricInterface will end up with a provider list where each provider in that list will keep the highest priority (lowest value) that has been encountered in iterating through fi_info output. There might be an edge case where `cxi+tcp` gets the same priority as a more performant provider on an interface on NUMA-1 (i.e. `ofi+verbs` on ib1 has priority 1 which would be the same as `cxi+tcp` on the same interface), but we can probably deal with this as a follow-on.

Reply:
Ah, I can push ucx+tcp lower if you like, although if we are using ucx, and it's possible for ofi to use verbs, I would expect ucx+dc/rc/ud to be available as well (and at highest priority so UCX would be preferred). One thing that is tricky is adjusting in relation to the ofi providers since they're in entirely different packages with internals hidden from each other.

Reply:
Adjusted the priority down. Let's see if this causes any issues.